### PR TITLE
feat(federation): unified inbox ingest log for follow-backfill

### PR DESCRIPTION
**Pull request by stephenhoward**
## Motivation

When a calendar owner followed a remote calendar, only events the source published *after* the follow surfaced — historical activity was invisible. The first attempt at follow-backfill (parked on a separate branch) shipped a synthetic in-memory dispatch path that bypassed `ap_inbox` entirely, which made it impossible to represent `Update` or `Undo` activities during backfill (`Undo` had no persisted target row to reference) and entangled backfill behavior with the dispatch implementation via a `.update` monkey-patch.

The fix is conceptual rather than mechanical: reframe `ap_inbox` from "the place signed POSTs land" to "the authenticated-activity log." Every row records the authentication mechanism that produced it, so backfilled activities and live HTTP-signed activities share one storage model and one dispatch pipeline. The chronological-dispatch invariant the table now enforces is what makes strict `Undo` cross-checks safe.

The architectural decision is recorded as DEC-011 in `agent-os/product/decisions.md`.

## Approach

**Schema and live ingest path (first commit).** Adds `auth_source` (`NOT NULL DEFAULT 'http_signature'`) and `auth_origin` (nullable) to `ap_inbox` via migration 0034, on `ActivityPubInboxMessageEntity` only — outbox is unaffected. `addToInbox` requires `InboxAuthContext` so future callers cannot accidentally elide the trust signal. The live HTTP route extracts the verified `keyId` origin from the signature middleware (not `req.headers.host`, which is the local server's own host) and falls back to `null` if the keyId is unparseable — the row still writes because the signature was middleware-verified. Both columns are excluded from `toModel()` and any API response.

**Backfill rewrite (second commit).** A pg-boss-queued worker walks the source outbox page-by-page through a signed GET, applies pre-write trust gates (`actor`-origin, `attributedTo` match for `Create`/`Update`, loop guard), and persists surviving activities as real `ap_inbox` rows tagged `auth_source='outbox_pull'`. After all pages have been persisted, a single `processInboxMessages` drain dispatches the rows in `message_time ASC` order through the existing inbox handlers. `findOrCreate` keeps the insert idempotent and first-writer-wins, so re-running backfill produces no duplicates and never overwrites a live HTTP-signed row's auth fields. A pure `clampMessageTime` helper bounds `message_time` to `[now - 2y, now + 1h]` to limit attacker influence on dispatch order.

**Dispatch-time strict gates (second commit).** Chronological dispatch makes two new gates safe to enforce at handler time rather than at write time:

- **`Undo` cross-check** in `processUnshareEvent`: the referenced `Announce` row must exist in `ap_inbox` with a non-null `auth_origin` whose host matches the `Undo` actor's host. Missing row, null `auth_origin`, host mismatch, or missing `undoActor` rejects the `Undo` silently without share-row mutation. Only local in-process dispatch (`trustLocalOrigin`) bypasses the check, since row provenance establishes trust there.
- **`Update` no-op** when target missing: if `Update(Event)` cannot resolve a target `ap_event_object`, the handler returns without mutation or `eventUpdated` emission. Prevents `Update`-after-`Delete` resurrection via backdated `published` timestamps.

**Cross-domain wiring.** `HousekeepingInterface` gains a `publishJob` method so the AP domain enqueues background work without importing pg-boss machinery directly. `ActivityPubDomain` receives `HousekeepingInterface` in its constructor; `ActivityPubEventHandlers` requires it so the follow-accepted backfill job cannot be silently dropped. `HousekeepingDomain` is constructed before `ActivityPubDomain` in `server.ts` so the dependency is satisfied at startup. `SyncRateLimiter` moves to `src/server/common/helper/` so the AP backfill worker and ICS sync share one sliding-window primitive (`src/server/calendar/service/import/sync.ts` keeps a re-export shim for existing callers).

**Incidental production-correctness fixes.** Two latent runtime bugs were caught by the federation E2E run and fixed in the second commit:

- `processInboxMessages` and `processOutboxMessages` drain queries used camelCase column aliases (`processedAt`, `messageTime`) that the entity decorators do not declare; the underlying columns are `processed_time` and `message_time`. Sequelize threw at runtime and the surrounding `finally` block swallowed it, leaving rows undrained. Both queries are now aligned with the schema.
- `ActivityPubEventHandlers` was constructed with an optional `HousekeepingInterface` left undefined in production, so `handleFollowAccepted` silently no-op'd on the guard. The interface is now required and wired through.

## Validation

- **31 unit tests** for the pure trust-gate helpers (`passesTrustGates`, `extractObjectId`) covering every branch — missing actor, malformed URLs, origin mismatch, loop guard, `Create`/`Update` `attributedTo` cases, bare-IRI hostname check, `Announce` origin variants, `Undo`/`Delete` pass-through, unknown types.
- **9 unit tests** for `clampMessageTime` covering within-window passthrough, floor and ceiling clamps, exact-boundary cases, `undefined`/`null`/empty/malformed inputs.
- **5 unit tests** for `extractKeyIdOrigin` covering happy path, fragment stripping, missing `Signature` header, non-URL `keyId`, missing `keyId` parameter.
- **10 follow-backfill integration tests** covering persistence shape (auth columns, clamped `message_time`), trust-gate rejection (forged `attributedTo`, foreign origin), drain ordering (shuffled `published` dispatched `ASC`), single-drain semantics across multi-page pagination, idempotency under re-run, first-writer-wins race against an existing HTTP-signed row, no `inboxMessageAdded` emission, SSRF rejection on private-IP source URL, and DEC-008 per-calendar dismissal scoping for backfilled `Undo`.
- **2 integration tests** for the live HTTP-signed POST writing correct `auth_source`/`auth_origin` (happy path + degraded null-`auth_origin` path for unparseable `keyId`).
- **3 federation E2E `test()` blocks** for base round-trip, `Update` round-trip, and `Undo` round-trip. The `Undo` block passes end-to-end against the federation Docker harness, validating the strict cross-check across a real cross-instance signed POST + backfill. The base and `Update` blocks fail in the current `docker-compose.federation.yml` harness because that harness runs only the web container; pg-boss jobs have no consumer. A follow-up will either add a worker container or run an in-process worker for non-production modes.
- `npm run lint`: 0 errors.
- 108/108 ActivityPub integration tests pass after the cross-domain wiring changes.
- `npm run build`: clean.

The two affected production methods (`processInboxMessages`, `processOutboxMessages`) and the `handleFollowAccepted` path were untestable by within-instance integration tests because those tests stub the drain to a no-op. The federation E2E surfaced both bugs.

### CODE REVIEW

File: src/server/activitypub/helper/remote-fetch.ts [hunk 1]
```diff
@@ -16,13 +20,96 @@ const USER_AGENT = 'Pavillion ActivityPub Server';
...
+ * @param calendar - The local calendar whose actor keypair signs the request
+ */
+async function buildSignedGetHeaders(
+  uri: string,
+  calendar: Calendar,
+): Promise<{ Signature: string; Date: string } | null> {
+  try {
+    // The outbox uses an EventEmitter-wired CalendarInterface; for read-only
+    // actor lookup we don't need a live event bus, so a fresh emitter is fine.
+    const calendarService = new CalendarInterface(new EventEmitter());
```
Comment (stephenhoward) [inline src/server/activitypub/helper/remote-fetch.ts]: Does using a brand new event emitter here disconnect any event emission/ handling from our main event bus? Would it be better to hand the helper the global event bus that all the domains are using?

File: src/server/activitypub/service/backfill.ts [hunk 2]
```diff
@@ -0,0 +1,681 @@
...
+ * activity, and persists survivors as real `ap_inbox` rows tagged with
+ * `auth_source = 'outbox_pull'` and `auth_origin = <source actor origin>`.
+ * After pagination terminates the worker calls
+ * `activityPubInterface.processInboxMessages()` exactly once to drain the
+ * inbox in chronological order — the same code path the live HTTP inbox
+ * runs after a signed POST.
+ *
+ * Shape B vs Shape A
+ * ------------------
+ * The previous (pv-cug3) implementation routed each activity through an
```
Comment (stephenhoward) [inline src/server/activitypub/service/backfill.ts]: Talking about development history in the comments is not useful. Comments should discuss what IS, not what WAS.

File: src/server/activitypub/helper/remote-fetch.ts (hunk 1 above)
Comment (stephenhoward) [reply src/server/activitypub/helper/remote-fetch.ts]: Investigated — wiring the global bus here would change nothing, so I corrected the comment instead (9c6f99e).

Two reasons it's a non-issue on this path:
- `CalendarInterface`'s constructor only *passes* the emitter to its services; it registers no listeners. So a throwaway emitter orphans nothing — there are no handlers to disconnect, and no per-call listener leak.
- The methods used here (`getActorByCalendarId`, `signActivity`) read `CalendarActorEntity` directly and never emit or touch the bus. The `CalendarInterface` is purely a constructor formality for `CalendarActorService` on this path.

I considered making `CalendarActorService`'s interface arg optional to drop the construction entirely, but that exposes `getActorByUrlName`/`getLocalCalendarByActorUri` (which *do* use it) to null-deref across 4 other call sites — higher risk for no behavioral gain. Left as a corrected comment.

File: src/server/activitypub/service/backfill.ts (hunk 2 above)
Comment (stephenhoward) [reply src/server/activitypub/service/backfill.ts]: Fixed in 9c6f99e — dropped the Shape A/Shape B framing and the "previous implementation" narrative from the worker doc comment, and removed the matching "Shape B" references and stale tracker tags from the sibling integration test and the trust-posture section. The comments now describe what the code is. Left the "rather than X" lines that contrast current behavior with alternatives, since those explain present design rather than history.

File: src/server/activitypub/helper/remote-fetch.ts (hunk 1 above)
Comment (stephenhoward) [reply src/server/activitypub/helper/remote-fetch.ts]: Update: superseding my reply above. Rather than leave it as a documented invariant, I implemented the threading fix in 9ac2904.

`buildSignedGetHeaders` no longer fabricates `new CalendarInterface(new EventEmitter())`. It now takes a `SigningContext { calendar, calendarInterface }` and reads the actor through the caller's bus-wired interface; `FollowBackfillService` passes the interface it already holds. `fetchRemoteObject`'s second arg became `signingContext?` accordingly; unsigned callers (inbox object fetches) are unchanged.

Reasoning for doing it rather than just documenting: the throwaway emitter was inert today but a latent foot-gun — a future emit on the signing path would be silently dropped on the dead bus with no error or log. Threading the live interface means any future emission reaches real subscribers. I avoided the alternative (making CalendarActorService's interface arg optional), which would have traded silent-loss for a null-deref risk across its other call sites.

Verified: lint clean, build clean, 19/19 remote-fetch unit + 10/10 follow-backfill integration tests pass.
